delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/Storages.scala

```diff
@@ -242,11 +242,13 @@ final class Storages private (
     * @param project
     *   the project where the storage belongs
     */
-  def fetch[R](
+  def fetch[R <: Throwable](
       resourceRef: ResourceRef,
       project: ProjectRef
-  )(implicit rejectionMapper: Mapper[StorageFetchRejection, R]): IO[R, StorageResource] =
-    fetch(IdSegmentRef(resourceRef), project).mapError(rejectionMapper.to)
+  )(implicit rejectionMapper: Mapper[StorageFetchRejection, R]): IO[StorageResource] =
```

Comment:
Do we still need this `R` and mapper? Looks like all it does is wrap the exception in a `WrappedStorageRejection` and is called only from `Files`

Reply:
Possibly not, but I've been trying to just migrate rather than change the logic. I wonder if @imsdu might know if this is safe

Reply:
We should try to remove those but this can be done in later PRs (unless they are blocking in some way), this one is gonna be complex enough.

Reply:
Can we open a ticket for it ?
So that we can address that after the current migration ?

Reply:
https://github.com/BlueBrain/nexus/issues/4441